### ISSUE TITLE
Bug in mcrals when adding unimod constraint?

### ISSUE THREAD

**Issue by gongyh** [svkucheryavski/mdatools#118]
When i add unimod constraint for the spec, error occurred as follows:
Error in if (y[i] <= max) { : argument is of length zero

My code is as follows:
```
fmcr <- mcrals(data, 100,
  cont.constraints = list(constraint("nonneg")),
  spec.constraints = list(
    constraint("nonneg"),
    constraint("unimod", params = list(tol = 0.05)),
    constraint("norm", params = list(type = "area"))
  ), verbose = TRUE
)
```

Is this a bug?

**Comment by svkucheryavski** [comment]
Hi, the second positional parameter by default is the expected number of pure components. I believe you wanted to have maximum 100 iterations right? If yes, just use:

```r
fmcr <- mcrals(data, ncomp,   max.niter = 100,...
```

Where `ncomp` is the number of components to resolve.

**Comment by gongyh** [comment]
Thanks for your help. I just want to decompose my signals into 100 pure unimod components.
When i debugged the constraintUnimod function yesterday, i found that errors occured when the component's peak locates at the first position (maybe also the last) during optimization.
For example, when the peak position is 1, then indseq = left_part = c(0, 1). In fact, indseq should not be 0.

**Comment by svkucheryavski** [comment]
Thank you very much for spotting this bug and for the pull request.

Here is a simple code which reproduces it:

```r
x  <- 1:500
y1 <- dnorm(x, m = 1, s = 20) * 0.7 + dnorm(x, m = 500, s = 20) * 0.3 ## main peak at the first column
y2 <- dnorm(x, m = 1, s = 20) * 0.3 + dnorm(x, m = 500, s = 20) * 0.7 ## main peak at the last column
y <- cbind(y1, y2)

cn <- constraint("unimod")
yp <- employ.constraint(cn, y, NULL)
```

I will check and accept the PR later today

**Comment by svkucheryavski** [comment]
closed in upcoming 0.14.2
